lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -576,34 +669,20 @@ def _on_relation_changed(self, event):
             return
 
         # TODO Future work, see if needed to check for all units restarted / primary elected
-        operator_password = self.database_requires.fetch_relation_field(
-            event.relation.id, OPERATOR_PASSWORD_KEY
-        )
-        backup_password = self.database_requires.fetch_relation_field(
-            event.relation.id, BACKUP_PASSWORD_KEY
-        )
+        (operator_password, backup_password) = self.get_cluster_passwords(event)
         if not operator_password or not backup_password:
             event.defer()
             self.charm.unit.status = WaitingStatus("Waiting for secrets from config-server")
             return
 
-        try:
-            self.update_password(
-                username=OperatorUser.get_username(), new_password=operator_password
-            )
-            self.update_password(BackupUser.get_username(), new_password=backup_password)
-        except RetryError:
-            self.charm.unit.status = BlockedStatus("Shard not added to config-server")
-            logger.error(
-                "Shard could not be added to config server, failed to set operator password."
-            )
-            event.defer()
-            return
+        self.sync_cluster_passwords(event, operator_password, backup_password)
 
         # after updating the password of the backup user, restart pbm with correct password
         self.charm._connect_pbm_agent()
         self.charm.app_peer_data["mongos_hosts"] = json.dumps(self.get_mongos_hosts())
 
+        self.charm.unit.status = ActiveStatus("")
```

Comment:
It looks like don't matter what status we had before; we changed it to Active here.
In any case, status handling up to Mehdi

Reply:
Hmm yes I agree, I will revert this change

Reply:
Yes, in general, i believe we should move away from setting `ActiveStatus` to rather "clearing" the specific statuses we're targeting. ActiveStatus is too intrusive in the sense that it overrides any component's status (even though unrelated with the current)